azure-iot-device/azure/iot/device/provisioning/internal/request_response_provider.py

```diff
@@ -0,0 +1,80 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+import logging
+import json
+from ..models.registration_result import RegistrationResult, RegistrationState
+
+logger = logging.getLogger(__name__)
+
+
+class RequestResponseMachine(object):
```

Comment:
>RequestResponseMachine [](start = 6, length = 22)

I don't really understand the point of this class. The only addition it seems to provide is a callback for receiving, which I'm not sure is enough to justify it's own class.

Then this name indicates there's a machine in here... but I don't see one?

Reply:
i think I have already changed the name of this class to RequestResponseProvider coz the machine name was misleading. those changes are not pushed. 
There are some more changes that needs to be done to this class based on Bert and my discussion but that will happen eventually.

---
In reply to: [277835502](https://github.com/Azure/azure-iot-sdk-python-preview/pull/81#discussion_r277835502) [](ancestors = 277835502)

Reply:
I will resolve this comment as of now

---
In reply to: [277896438](https://github.com/Azure/azure-iot-sdk-python-preview/pull/81#discussion_r277896438) [](ancestors = 277896438,277835502)